src/main/java/RunnableThead.java

```diff
@@ -0,0 +1,18 @@
+import org.apache.log4j.Logger;
+
+public class RunnableThead implements Runnable {
+    public static final int LIMIT = 1000;
```

Comment:
```suggestion
    public static final int LIMIT = 100;
```
100 is enough

Reply:
It isn't enough for my PC. It quite offen runs the race for one thread. So to verify I increased the threshold